scout/load/panelapp.py

```diff
@@ -0,0 +1,132 @@
+            hgnc_symbol_to_ensembl_id_map=hgnc_symbol_to_ensembl_id_map,
+            institute=institute,
+            confidence=confidence,
+        )
+
+        if len(parsed_panel["genes"]) == 0:
+            LOG.warning("Panel %s is missing genes. Skipping.", parsed_panel["display_name"])
+            continue
+
+        adapter.load_panel(parsed_panel=parsed_panel, replace=True)
+
+
+def get_panelapp_genes(
+    adapter: MongoAdapter, institute: str, panel_ids: List[str], types_filter: List[str]
+) -> Set[tuple]:
+    """Parse and collect genes from one or more panelApp panels."""
+
+    genes = set()
+    ensembl_id_to_hgnc_id_map: Dict[str, int] = adapter.ensembl_to_hgnc_id_mapping()
+    hgnc_symbol_to_ensembl_id_map: Dict[int, str] = adapter.hgnc_symbol_ensembl_id_mapping()
+
+    with progressbar(panel_ids, label="Parsing panels", length=len(panel_ids)) as panel_ids:
+        for panel_id in panel_ids:
+            panel_dict: dict = panelapp.get_panel(panel_id)
+            panel_type_slugs = [type["slug"] for type in panel_dict.get("types")]
+            # Parse panel only if it's of the expect type(s)
+            if not set(types_filter).intersection(panel_type_slugs):
+                continue
+
+            parsed_panel = parse_panelapp_panel(
+                panel_info=panel_dict,
+                ensembl_id_to_hgnc_id_map=ensembl_id_to_hgnc_id_map,
+                hgnc_symbol_to_ensembl_id_map=hgnc_symbol_to_ensembl_id_map,
+                institute=institute,
+                confidence="green",
+            )
+            genes.update(
+                {(gene["hgnc_id"], gene["hgnc_symbol"]) for gene in parsed_panel.get("genes")}
+            )
+
+    return genes
+
+
+def load_panelapp_green_panel(adapter: MongoAdapter, institute: str, force: bool, signed_off: bool):
+    """Load/Update the panel containing all Panelapp Green genes."""
+
+    def parse_types_filter(types_filter: str, available_types: List[str]) -> List[str]:
+        """Translate panel type input from users to panel type slugs."""
+        if not types_filter:
+            return available_types
+        index_list = [int(typeint) - 1 for typeint in types_filter.replace(" ", "").split(",")]
+        return [available_types[i] for i in index_list]
+
+    # check and set panel version
+    old_panel = adapter.gene_panel(panel_id=PANEL_NAME)
+    green_panel = {
+        "panel_name": PANEL_NAME,
+        "display_name": "PanelApp Green Genes",
+        "institute": institute,
+        "version": float(math.floor(old_panel["version"]) + 1) if old_panel else 1.0,
+        "date": datetime.now(),
+    }
+
+    LOG.info("Fetching all PanelApp panels")
+
+    panel_ids = panelapp.get_panel_ids(signed_off=signed_off)
+    LOG.info(f"\n\nQuery returned {len(panel_ids)} panels\n")
+    LOG.info("Panels have the following associated types:")
+    available_types: List[str] = panelapp.get_panel_types()
```

Comment:
Could we do this on the CLI as well? I don't mind an `input()` fallback, but it is Good Practice to be able to run things in batch without having to rely on manual input. We could even have the panel types fetched for the `--help` output or a separate subcommand if you feel they are liable to change.

Reply:
We could, but I had this design in mind for this reason: 
If we had a list of possible panel types, say in the constants, we could use them as options in the cli at the beginning of the command, as you suggest. Butwhat if they introduce a new model type? Then we wouldn't be aware for quite some time (because we don't follow these things on PanelApp). 
Instead panel types get collected in real time from panel data and the options presented are always going to be up-to-date.

One thing I'd do for simplifying the options is perhaps ordering them so that the options will follow always the same order when they are presented to the user 🤔

Reply:
Ordering is good! How about trying for dynamically checking the choice list anyway? It would not likely change between a call to retrieve the list and a second call. Having the user listing clear text category names on the command line/option would also add some safety.

Reply:
mmm I don't like much the idea of having the user to write text, since it can be a lot of categories and if you start to add more than one it can be really cumbersone. There are already two checks:

After the input when you write for instance 1,3,5, the script will list the chosen categories. The second check will be the description of the panel, where it clearly states the types of panel that were used for collecting the genes

Reply:
Mm, in particular any options like this should be possible to add to config. How about that then, a dict with (cleartext) options in the config.py to configure what will happen by default. If none is given, or it is overridden, you could use the number input thing?

Reply:
Seems like an ovecomplication to me. Most facilities won't use this functionality. Lund will just press enter lo use all panels. When it comes to us, if we want to keep using the same settings as before we will insert one number. Otherwise a list of numbers, like 3 or 4 perhaps? And if we sort the options it will be likely the same sumber all over. Config file is already complicated as it is, and don't forget we are deprecating the config file currently used by the cli